packages/plugin-degenfans-alfafrens/src/utils/degenfansApi.ts

```diff
@@ -0,0 +1,157 @@
+      };
+    } else {
+      headers = {
+        'Accept': 'application/json'
+      };
+    }
+    const response = await fetch(degenfansApiBaseUrl + url, {
+      method,
+      headers,
+      body
+    });
+
+    // Check if the response status is OK (status code 200-299)
+    if (!response.ok) {
+      throw new Error(`HTTP error! Status: ${response.status}`);
+    }
+
+    // Parse the JSON response
+    const apiData = await response.json() as DegenFansResponse<T>;
+
+    return apiData;  // Return the parsed data
+  } catch (error) {
+
+    if (error instanceof Error) {
+      return { message: error.message, status: 500 } as DegenFansResponse<null>;
+    } else {
+      return { message: 'Unexpected error:', status: 500 } as DegenFansResponse<null>;
+    }
+    // You can also return null or a default value in case of an error
+
+  }
+}
+function callPostDfApi<T>(url: string, body: string) {
+  return callDfApi<T>(url, Method.POST, body);
+}
+function callGetDfApi<T>(url: string) {
+  return callDfApi<T>(url, Method.GET, null);
+}
+
+export function getHelpText(user: AfUser): string {
+  let tbl = "";
+  if (user && user.matchType) {
+    if (user.matchType === "BY_CREATOR_COIN") {
+      tbl += "\n* I matched your AlfaFrens user by your moxie creator coin FID";
+    } else if (user.matchType === "BY_GIVEN_ADDRESS") {
+      tbl += "\n* I matched your AlfaFrens user by your given AlfaFrens user address";
+    } else if (user.matchType === "BY_GIVEN_NAME") {
+      tbl += "\n* I matched your AlfaFrens user by your given AlfaFrens user name";
+    } else if (user.matchType === "BY_FID") {
+      tbl += "\n* I matched your AlfaFrens user by your moxie account FID";
+    } else if (user.matchType === "BY_TWITTER") {
+      tbl += "\n* I matched your AlfaFrens user by your moxie account X handle";
+    }
+  } 
+  return tbl
+}
+export function getHelpTextUserNotFound():string{
+  let tbl="";
+  tbl += "I was not able to find your AlfaFrens profile. To get a personalized staking recommendations, make sure that you have:";
+  tbl += "\n  * AlfaFrens profile connected to your Farcaster and X account";
+  tbl += "\n  * Conected Farcaster Account from your Moxie profile";
+  tbl += "\n  * Conected X Account from your Moxie profile";
+  tbl += "\n";
+  tbl += "\nIf you don't have an account on AlfaFrens, you can create one on:";
+  tbl += "\n[https://alfafrens.com](https://alfafrens.com)";
+  tbl += "\n\nElse, you can get in touch with @[degenfans|M155] or [AlfaFrens - DegenFans](https://www.alfafrens.com/channel/0x997a65d6e7beafd3ea548b499fd2ea7a7875721f) to resolve the issue.\n";
+  return tbl;
+}
+
+export interface UserData {
+  fid: string, xhandle: string
+}
+
+export function getUserData(moxieUserInfo: MoxieUser): UserData {
+  let fid: string = null;
+  let xhandle: string = null;
+  const fcId = moxieUserInfo.identities.find(o => o.type === 'FARCASTER');
+  if (fcId) {
+    fid = (fcId.metadata as FarcasterMetadata).profileTokenId;
```

Comment:
Minor fix here, seems the data structure in mock data previously is not right, correcting it here

```suggestion
    fid = (fcId.metadata as FarcasterMetadata)?.fid?.toString();
```